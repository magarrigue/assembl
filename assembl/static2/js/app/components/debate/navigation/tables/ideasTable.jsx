// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { graphql, compose } from 'react-apollo';

import AllIdeasQuery from '../../../../graphql/AllIdeasQuery.graphql';
import MenuList, { type ItemNode } from './menuList';
import withLoadingIndicator from '../../../common/withLoadingIndicator';

type IdeasTableProps = {
  identifier: string,
  phaseId: string,
  onMenuItemClick: Function,
  data: {
    loading: boolean,
    error: Object,
    ideas: Array<ItemNode>,
    rootIdea: ItemNode
  }
};

export function DumbIdeasTable(props: IdeasTableProps) {
  const { identifier, phaseId, onMenuItemClick, data } = props;
  const { ideas, rootIdea } = data;
  return (
    <MenuList
      items={ideas}
      rootItem={rootIdea && rootIdea.id}
      phaseId={phaseId}
      identifier={identifier}
      onMenuItemClick={onMenuItemClick}
    />
  );
}

const IdeasTableWithData = graphql(AllIdeasQuery);

const mapStateToProps = state => ({
  lang: state.i18n.locale,
  debate: state.debate
});

export default compose(connect(mapStateToProps), IdeasTableWithData, withLoadingIndicator())(DumbIdeasTable);