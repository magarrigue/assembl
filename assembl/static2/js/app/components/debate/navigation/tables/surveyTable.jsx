// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { graphql, compose } from 'react-apollo';
import DebateThematicsQuery from '../../../../graphql/DebateThematicsQuery.graphql';
import MenuList, { type ItemNode } from './menuList';
import withLoadingIndicator from '../../../common/withLoadingIndicator';

type SurveyTableProps = {
  identifier: string,
  phaseId: string,
  onMenuItemClick: Function,
  data: {
    loading: boolean,
    error: Object,
    thematics: Array<ItemNode>
  }
};

export function DumbSurveyTable(props: SurveyTableProps) {
  const { identifier, phaseId, onMenuItemClick, data } = props;
  const { thematics } = data;
  return <MenuList items={thematics} phaseId={phaseId} identifier={identifier} onMenuItemClick={onMenuItemClick} />;
}

const SurveyTableWithData = graphql(DebateThematicsQuery);

const mapStateToProps = state => ({
  lang: state.i18n.locale,
  debate: state.debate
});

export default compose(connect(mapStateToProps), SurveyTableWithData, withLoadingIndicator())(DumbSurveyTable);