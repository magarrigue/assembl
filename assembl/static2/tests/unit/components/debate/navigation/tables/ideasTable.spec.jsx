import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { DumbIdeasTable } from '../../../../../../js/app/components/debate/navigation/tables/ideasTable';

describe('IdeasTable component', () => {
  it('should match the IdeasTable', () => {
    const data = {
      ideas: [
        {
          id: 'foo',
          parentId: 'root'
        },
        {
          id: 'bar',
          parentId: 'root'
        }
      ],
      rootIdea: {
        id: 'root'
      }
    };
    const props = {
      identifier: 'multiColumns',
      phaseId: 'phaseFoo',
      data: data
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DumbIdeasTable {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});