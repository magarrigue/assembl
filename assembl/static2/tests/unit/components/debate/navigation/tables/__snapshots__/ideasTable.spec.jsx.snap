// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeasTable component should match the IdeasTable 1`] = `
<MenuList
  className=""
  identifier="multiColumns"
  items={
    Array [
      Object {
        "id": "foo",
        "parentId": "root",
      },
      Object {
        "id": "bar",
        "parentId": "root",
      },
    ]
  }
  onMenuItemClick={undefined}
  phaseId="phaseFoo"
  rootItem="root"
/>
`;
